src/CommandLine.cpp

```diff
@@ -133,7 +133,7 @@ void sub_612910(std::int32_t argc, char **argv) {
             for (std::uint32_t j = 0; j < y->count; ++j) {
                 y->args[j] = MC2_STRDUP(argv[i + j]);
             }
-            i = x;
+            i = x - 1;
```

Comment:
It definitely should be x - 1 as otherwise the next command line argument will not be processed. In the original code this would have just jumped to the next loop but because of the ++i you need to decrement it.